### PR TITLE
Use optimized Bullet raytest function in CharacterGhostObject::rayTest()

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/15596/Should-use-btGhostObject-rayTest-in-CharacterGhostObject-rayTest-for-optimization
This PR makes a small change to the code responsible for avatar collision checking in order to improve performance. The change should not affect program behavior.

## TEST PLAN

* Make sure that when an avatar is hovering above an object, that the height at which they transition to falling is the same
* Make sure that when an avatar walks or flies into an object, that the height relative to the avatar height at which movement is blocked remains the same, and detects thin objects (such as fences) with the same level of reliability
* Make sure that when an avatar walks into an object that does not block motion, the avatar is able to ascend on top of the object
* Make sure that when an avatar flies into an object that does not block motion, the avatar is able to ascend on top of the object, and continues flying as before (ie does not transition to walking)
* Make sure that the avatar is only able to walk up slopes below the same level of steepness

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by sethalves** [comment]
add to whitelist

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10361/HighFidelity-Beta-PR13304-9ac1443.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10361/HighFidelity-Beta-Interface-PR13304-9ac1443.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10361/HighFidelity-Beta-PR13304-9ac1443.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10361/HighFidelity-Beta-Interface-PR13304-9ac1443.exe

**Comment by sabrina-shanman** [comment]
I'm closing this for now until I can better explain this and its implications.

**Comment by sabrina-shanman** [comment]
I have tested moving an avatar into cubes of various heights, and see the same behavior before and after the change, ie the player avatar cannot walk onto the cube unless it is below a certain height. Since this behavior depends on CharacterGhostObject::rayTest(), that shows the code change did not affect functionality.

On a test branch, I placed profiling code directly around the line of code before and after the change, and took a running average of the time difference. I then took a cube entity creation script provided by Andrew, which spawns a grid of cubes around the player, and modified the script to produce 40*40*40 (64000) cubes.

The cube entity creation script was run in the default development world with the blank skybox and the white platform, and I controlled the avatar to make it either walk on the platform or fly between the cubes. Distances between the cubes were roughly equal to the height of the avatar.

Results of the profiling before and after the change (average execution time over region of code changed, in nanoseconds):
Before (walking on platform): 576 ns
Before (flying): 616 ns
After (walking on platform): 545 ns
After (flying): 53 ns

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10480/HighFidelity-Beta-PR13304-9ac1443.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10480/HighFidelity-Beta-PR13304-9ac1443.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10480/HighFidelity-Beta-Interface-PR13304-9ac1443.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10480/HighFidelity-Beta-PR13304-9ac1443.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10480/HighFidelity-Beta-Interface-PR13304-9ac1443.exe

**Comment by sabrina-shanman** [comment]
This change should not affect program behavior.

Test plan:

* Make sure that when an avatar is hovering above an object, that the height at which they transition to falling is the same
* Make sure that when an avatar walks or flies into an object, that the height relative to the avatar height at which movement is blocked remains the same, and detects thin objects (such as fences) with the same level of reliability
* Make sure that when an avatar walks into an object that does not block motion, the avatar is able to ascend on top of the object
* Make sure that when an avatar flies into an object that does not block motion, the avatar is able to ascend on top of the object, and continues flying as before (ie does not transition to walking)
* Make sure that the avatar is only able to walk up slopes below the same level of steepness

**Comment by sabrina-shanman** [comment]
btPairCachingGhostObject didn't seem to be doing anything useful, so CharacterGhostObject now uses the base btGhostObject.

Quick profiling to make sure:
Walking: 488ns
Flying: 49ns

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10523/HighFidelity-Beta-PR13304-d4d8f36.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10523/HighFidelity-Beta-PR13304-d4d8f36.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10523/HighFidelity-Beta-Interface-PR13304-d4d8f36.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10523/HighFidelity-Beta-PR13304-d4d8f36.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10523/HighFidelity-Beta-Interface-PR13304-d4d8f36.exe

**Comment by sabrina-shanman** [comment]
Remove labels in light of stale review.

File: libraries/physics/src/CharacterGhostObject.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ bool CharacterGhostObject::rayTest(const btVector3& start,
         const btVector3& end,
         CharacterRayResult& result) const {
     if (_world && _inWorld) {
-        _world->rayTest(start, end, result);
+        this->btGhostObject::rayTest(start, end, result);
```
Comment (AndrewMeadows) [inline libraries/physics/src/CharacterGhostObject.cpp]: Remove the unnecessary `this->`.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10589/HighFidelity-Beta-PR13304-baf0df0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10589/HighFidelity-Beta-Interface-PR13304-baf0df0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10589/HighFidelity-Beta-PR13304-baf0df0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10589/HighFidelity-Beta-Interface-PR13304-baf0df0.exe

**Comment by Qliemillar** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10657/HighFidelity-Beta-PR13304-baf0df0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10657/HighFidelity-Beta-PR13304-baf0df0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10657/HighFidelity-Beta-Interface-PR13304-baf0df0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10657/HighFidelity-Beta-PR13304-baf0df0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10657/HighFidelity-Beta-Interface-PR13304-baf0df0.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Tested on all platforms, animations look good and height for falling looks as expected.

**Pull request merged**